music21/meter.py

```diff
@@ -3786,25 +3784,19 @@ def fixBeamsOneElementDepth(i, el, depth):
                 else:
                     beamType = 'start'
 
-
-            # last beams was active, last beamNumber was active,
-            # and it was stopped or was a partial-left
             elif (beamPrevious is not None
                   and beamNumber in beamPrevious.getNumbers()
-                  and beamPrevious.getTypeByNumber(beamNumber) in ['stop', 'partial-left']
-                  and beamNext is not None):
-                beamType = 'start'
-
-
-            # last note had beams but stopped, next note cannot be beamed to
-            # was active, last beamNumber was active,
-            # and it was stopped or was a partial-left
-            elif (beamPrevious is not None
-                  and beamNumber in beamPrevious.getNumbers()
-                  and beamPrevious.getTypeByNumber(beamNumber) in ['stop', 'partial-left']
-                  and beamNext is None):
-                beamType = 'partial-left'  # will be deleted later in the script
-
+                  and beamPrevious.getTypeByNumber(beamNumber) in ['stop', 'partial-left']):
+                    # last beams was active, last beamNumber was active,
+                    # and it was stopped or was a partial-left
+                    if beamNext is not None:
+                        beamType = 'start' if beamNumber in beamNext.getNumbers() else 'partial-right'
+
+                    # last note had beams but stopped, next note cannot be beamed to
+                    # was active, last beamNumber was active,
+                    # and it was stopped or was a partial-left
+                    else:
+                        beamType = 'partial-left'  # will be deleted later in the script
```

Comment:
Here I simply did 2 things:

1. merge the two identical long "if"s into one with additional nested "if-else". No code was changed here.
2. Fix the source of the beaming issues: change `beamType = 'start'` into `beamType = 'start' if beamNumber in beamNext.getNumbers() else 'partial-right'`